### PR TITLE
External input parsing

### PR DESCRIPTION
**Pull request by danielgranhao**
Addresses #535.

TODO:
- [x] Fix bindings
- [x] Docs

### CODE REVIEW

**Comment by danielgranhao** [comment]
I'm currently moving the external parsing logic to sdk-common.

**Comment by dangeross** [comment]
There are some docs here on how to regenerate the React Native bindings: 
https://github.com/breez/breez-sdk-liquid/blob/main/packages/react-native/DEVELOPING.md#prerequisites
Here are more specific versions needed to be installed locally to match the CI:
```
kotlin@2.1.0
swiftformat@0.55.3
ktlint@1.4.1
tslint@6.1.3
typescript@5.7.2
```
After running `make init` in the `packages/react-native` dir, you only need to run `make react-native` to regenerate the bindings

**Review by dangeross — COMMENTED** [review]
I'm wondering if the external input parsers could be set in the `Config` and passed during the `connect()`. Do you see that they could be changed during the life of the SDK instance?

**Comment by danielgranhao** [comment]
> I'm wondering if the external input parsers could be set in the Config and passed during the connect(). Do you see that they could be changed during the life of the SDK instance?

@dangeross Hm.. Indeed, I don't see that need. Using the config LGTM. Wdyt @roeierez?

**Comment by roeierez** [comment]
> > I'm wondering if the external input parsers could be set in the Config and passed during the connect(). Do you see that they could be changed during the life of the SDK instance?
> 
> @dangeross Hm.. Indeed, I don't see that need. Using the config LGTM. Wdyt @roeierez?

Yes I am fine with using the config

**Review by dangeross — APPROVED** [review]
LGTM

**Review by hydra-yse — APPROVED** [review]
utACK

File: lib/bindings/src/breez_sdk_liquid.udl [hunk 1]
```diff
@@ -335,6 +335,7 @@ dictionary Config {
     string? breez_api_key;
     string? cache_dir;
     u64? zero_conf_max_amount_sat;
+    sequence<ExternalInputParser>? external_input_parsers;
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply lib/bindings/src/breez_sdk_liquid.udl]: Applied in [3705175](https://github.com/breez/breez-sdk-liquid/pull/597/commits/3705175093c0aebed0bdcfc81949bb2d7e0786c8)

**Review by roeierez — APPROVED** [review]
LGTM

**Pull request merged**